src/ape/api/accounts.py

```diff
@@ -201,7 +202,7 @@ def check_signature(
         Returns:
             bool: ``True`` if the data was signed by this account. ``False`` otherwise.
         """
-        if isinstance(data, SignableMessage):
+        if isinstance(data, (SignableMessage, EIP712SignableMessage)):
```

Comment:
Hmm, I actually like this. I don't think it would be too badly in conflict if this was supported in core (because it has paths for handling other types of messages)